client/app/src/main/java/com/microsoft/research/karya/ui/scenarios/speechData/SpeechDataMainFragment.kt

```diff
@@ -0,0 +1,296 @@
+      recordBtn.setBackgroundResource(
+        when (state) {
+          DISABLED -> R.drawable.ic_mic_disabled
+          ENABLED -> R.drawable.ic_mic_enabled
+          ACTIVE -> R.drawable.ic_mic_enabled
+        }
+      )
+    }
+
+    viewModel.playBtnState.observe(viewLifecycleOwner.lifecycle, lifecycleScope) { state ->
+      playBtn.isClickable = state != DISABLED
+      playBtn.setBackgroundResource(
+        when (state) {
+          DISABLED -> R.drawable.ic_speaker_disabled
+          ENABLED -> R.drawable.ic_speaker_enabled
+          ACTIVE -> R.drawable.ic_speaker_enabled
+        }
+      )
+    }
+
+    viewModel.nextBtnState.observe(viewLifecycleOwner.lifecycle, lifecycleScope) { state ->
+      nextBtn.isClickable = state != DISABLED
+      nextBtn.setBackgroundResource(
+        when (state) {
+          DISABLED -> R.drawable.ic_next_disabled
+          ENABLED -> R.drawable.ic_next_enabled
+          ACTIVE -> R.drawable.ic_next_enabled
+        }
+      )
+    }
+
+    viewModel.sentenceTvText.observe(viewLifecycleOwner.lifecycle, lifecycleScope) { text ->
+      sentenceTv.text = text
+    }
+
+    viewModel.recordSecondsTvText.observe(viewLifecycleOwner.lifecycle, lifecycleScope) { text ->
+      recordSecondsTv.text = text
+    }
+
+    viewModel.recordCentiSecondsTvText.observe(viewLifecycleOwner.lifecycle, lifecycleScope) { text ->
+      recordCentiSecondsTv.text = text
+    }
+
+    viewModel.playbackProgressPb.observe(viewLifecycleOwner.lifecycle, lifecycleScope) { progress ->
+      playbackProgressPb.progress = progress
+    }
+
+    viewModel.playbackProgressPbMax.observe(viewLifecycleOwner.lifecycle, lifecycleScope) { max ->
+      playbackProgressPb.max = max
+    }
+
+    viewModel.playRecordPromptTrigger.observe(viewLifecycleOwner.lifecycle, lifecycleScope) { play ->
+      if (play) {
+        playRecordPrompt()
+      }
+    }
+
+  }
+
+  private fun playRecordPrompt() {
+    val oldColor = sentenceTv.currentTextColor
+
+    assistant.playAssistantAudio(
+      AssistantAudio.RECORD_SENTENCE,
+      uiCue = {
+        sentenceTv.setTextColor(Color.parseColor("#CC6666"))
+        sentencePointerIv.visible()
+      },
+      onCompletionListener = {
+        lifecycleScope.launch {
+          sentenceTv.setTextColor(oldColor)
+          sentencePointerIv.invisible()
+          delay(500)
+          playRecordAction()
+        }
+      }
+    )
+  }
+
+  private fun playRecordAction() {
+
+    lifecycleScope.launch {
+      assistant.playAssistantAudio(
+        AssistantAudio.RECORD_ACTION,
+        uiCue = {
+          recordPointerIv.visible()
+          recordBtn.setBackgroundResource(R.drawable.ic_mic_enabled)
+        },
+        onCompletionListener = {
+          lifecycleScope.launch {
+            recordPointerIv.invisible()
+            delay(500)
+            playStopAction()
+          }
+        }
+      )
+      delay(1500)
+      recordBtn.setBackgroundResource(R.drawable.ic_mic_active)
```

Comment:
If the app is closed in these 1.5 seconds or the user moves to the previous screen, it'll just crash the app. I think we have to fix this first.